File: backend/Domain/ApiModels/Artist/ArtistSearchResult.cs
using System.Collections.Generic;

namespace Domain.ApiModels.Artist
{
    public class ArtistSearchResult
    {
        public string ArtistName { get; set; }
        public string FormationYear { get; set; }
        public string DisbandYear { get; set; }
        public string NumberOfMembers { get; set; }
        public string Genre { get; set; }
        public string Biography { get; set; }
        public string LogoUrl { get; set; }
        public string ThumbnailUrl { get; set; }
        public string BandWebsiteUrl { get; set; }
        public string BrainzId { get; set; }

        public List<ArtistAlbumSearchResult> Albums { get; set; }
    }

    public class ArtistAlbumSearchResult
    {
        public string AlbumName { get; set; }
        public string AlbumType { get; set; }
        public string YearReleased { get; set; }
        public string ThumbnailUrl { get; set; }
        public string Description { get; set; }
        public string BrainzId { get; set; }
    }
}

File: frontend/client/src/types/ArtistSearchResult.ts
export interface ArtistSearchResult {
    artistName: string,
    formationYear: string,
    disbandYear: string|null,
    numberOfMembers: string,
    genre: string,
    biography: string,
    logoUrl: string,
    thumbnailUrl: string,
    bandWebsiteUrl: string,
    brainzId: string,
    albums: ArtistAlbumSearchResult[]
}

export interface ArtistAlbumSearchResult {
    albumName: string,
    albumType: string,
    yearReleased: string,
    thumbnailUrl: string,
    description: string,
    brainzId: string
}